dipy/workflows/segment.py

```diff
@@ -1,84 +1,85 @@
 from __future__ import division, print_function, absolute_import
 
-from glob import glob
-from os.path import join, basename, splitext
+import inspect
+import logging
+import numpy as np
 
 import nibabel as nib
-import numpy as np
 
-from dipy.workflows.utils import choose_create_out_dir
 from dipy.segment.mask import median_otsu
+from dipy.workflows.workflow import Workflow
+
+
+class MedianOtsuFlow(Workflow):
+    @classmethod
+    def get_short_name(cls):
+        return 'medotsu'
+
+
+    def run(self, input_files, save_masked=False, median_radius=2, numpass=5,
+            autocrop=False, vol_idx=None, dilate=None, out_dir='',
+            out_mask='brain_mask.nii.gz', out_masked='dwi_masked.nii.gz'):
+        """ Workflow wrapping the median_otsu segmentation method.
+
+        Applies median_otsu segmentation on each file found by 'globing'
+        ``input_files`` and saves the results in a directory specified by
+        ``out_dir``.
+
+        For a quick brain segmentation, use --numpass 5 and --median_radius 2
+
+        Parameters
+        ----------
+        input_files : string
+            Path to the input volumes. This path may contain wildcards to
+            process multiple inputs at once.
+        save_masked : bool
+            Save mask
+        median_radius : int, optional
+            Radius (in voxels) of the applied median filter(default 2)
+        numpass : int, optional
+            Number of pass of the median filter (default 5)
+        autocrop : bool, optional
+            If True, the masked input_volumes will also be cropped using the
+            bounding box defined by the masked data. Should be on if DWI is
+            upsampled to 1x1x1 resolution. (default False)
+        vol_idx : string, optional
+            1D array representing indices of ``axis=3`` of a 4D `input_volume`
+            'None' (the default) corresponds to ``(0,)`` (assumes first volume
+            in 4D array)
+        dilate : string, optional
+            number of iterations for binary dilation (default 'None')
+        out_dir : string, optional
+            Output directory (default input file directory)
+        out_mask : string, optional
+            Name of the mask volume to be saved (default 'brain_mask.nii.gz')
+        out_masked : string, optional
+            Name of the masked volume to be saved (default 'dwi_masked.nii.gz')
+        """
+        io_it = self.get_io_iterator()
+
+        for fpath, mask_out_path, masked_out_path in io_it:
+            logging.info('Applying median_otsu segmentation on {0}'.
+                         format(fpath))
+
+            img = nib.load(fpath)
+            volume = img.get_data()
+
+            masked_volume, mask_volume = median_otsu(volume, median_radius,
+                                                     numpass, autocrop, vol_idx,
+                                                     dilate)
+
+            mask_img = nib.Nifti1Image(mask_volume.astype(np.float32),
```

Comment:
I suggest to start using those functions that we added in io.image
```
from dipy.io.image import load_nifti, save_nifti
```
to reduce repetition and number of lines in the code.